ksql-tools/src/main/java/io/confluent/ksql/tools/printmetrics/PrintMetrics.java

```diff
@@ -0,0 +1,158 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.tools.printmetrics;
+
+import sun.tools.jconsole.LocalVirtualMachine;
+
+import javax.management.*;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class PrintMetrics {
+  public static void printHelp() {
+    System.err.println(
+        "usage: PrintMetrics "
+            + "[help] "
+            + "pid=<KSQL PID>");
+  }
+
+  private static void printMetrics(int pid) throws IOException {
+    // get all the jvms running locally and find the one we are interested in
+    Map<Integer, LocalVirtualMachine> jvms = LocalVirtualMachine.getAllVirtualMachines();
+    LocalVirtualMachine ksqlJvm = null;
+    if (pid == -1) {
+      for (LocalVirtualMachine jvm : jvms.values()) {
+        System.out.println(jvm.displayName());
+        if (jvm.displayName().startsWith("io.confluent.ksql.rest.server.KsqlRestApplication")) {
+          ksqlJvm = jvm;
+        }
+      }
+    } else {
+      ksqlJvm = jvms.get(pid);
+    }
+    if (ksqlJvm == null) {
+      throw new PrintMetricsException("Could not find vm with pid " + pid);
+    }
+
+    String jvmAddress = ksqlJvm.connectorAddress();
+    JMXServiceURL jmxURL;
+    try {
+      jmxURL =new JMXServiceURL(jvmAddress);
+    } catch(MalformedURLException e) {
+      throw new PrintMetricsException(String.format("No JVM at PID %d", pid));
+    }
+    JMXConnector connector = JMXConnectorFactory.connect(jmxURL);
+    connector.connect();
+    MBeanServerConnection connection = connector.getMBeanServerConnection();
+
+    // print out all ksql engine metrics
+    Set<ObjectName> names = connection.queryNames(null, null);
+    List<ObjectName> ksqlObjects = new LinkedList<>();
+    for (ObjectName n : names) {
+      if (n.toString().startsWith("io.confluent.ksql.metrics:type=ksql-engine")) {
+        ksqlObjects.add(n);
+      }
+    }
+    for (ObjectName n : ksqlObjects) {
+      MBeanInfo info;
+      try {
+        info = connection.getMBeanInfo(n);
+      } catch (Exception error) {
+        throw new PrintMetricsException("Unexpected error getting mbean info " + error.getMessage());
+      }
+      MBeanAttributeInfo[] attributes = info.getAttributes();
+      for (MBeanAttributeInfo attributeInfo : attributes) {
+        Object attribute;
+        try {
+          attribute = connection.getAttribute(n, attributeInfo.getName());
+        } catch (Exception error) {
+          throw new PrintMetricsException("Unexpected error getting attribute " + error.getMessage());
+        }
+        System.out.println(attributeInfo.getName() + ": " + attribute);
```

Comment:
at some point we might want to provide a formatter so data could displayed as CSV, JSON, statsd or delimited - the output could then be sent to a remote REST EP

Reply:
Missed this. I'll file an issue so we don't lose track of it.